Fix behavior when connecting wss url without TLS support

maugier opened the pull request:
Hi,

  When compiled without TLS support, the client currently falls back to HTTP (and by default, on port 443) even if the URI has the `wss` scheme. This creates a problem that is hard to debug, as the call fails with a  generic WSS protocol error.

  If a `wss://` URI is requested and no TLS support is available, the client should fail fast with a `UrlError::TlsFeatureNotEnabled`.

Cheers,

daniel-abramov reviewed (commented):
This looks like a useful thing! I think it would be even more awesome thing if we could cover it with a small test 🙂

maugier commented:
Writing the test made me realize that the failure should happen earlier, before we even attempt to connect the TCP stream.

I reverted the earlier patch and moved the check earlier. What do you think of this ?

daniel-abramov commented:
> I reverted the earlier patch and moved the check earlier. What do you think of this ?

Looks very sensible, thanks!

daniel-abramov commented:
So the only job that still fails does this only due to our low MSRV, as discussed in https://github.com/snapview/tungstenite-rs/pull/432 and other related issues, so I'll merge it and raise MSRV in a separate commit then.
